### ISSUE TITLE
Where clause not working in deleting records selected with subquery

### ISSUE THREAD

**Issue by zella** [orientechnologies/orientdb#6174]
## Expected behavior and actual behavior
I try to delete record selected with subquery, but 'WHERE' clause not working, it's ignores, all records removes instead:
` delete from (select expand(accounts) from #30:0) where type = 'test2' ` - removes all accounts from #30:0, 

However same with select: where working
` select from (select expand(accounts) from #30:0) where type = 'test2' `

## Steps to reproduce the problem

```
CREATE CLASS Account
CREATE CLASS Profile
CREATE PROPERTY Profile.accounts LINKLIST
Insert into Account set type = 'test1'
Insert into Account set type = 'test2'
Insert into Account set type = 'test3'
Insert into Profile set accounts = [#29:0,#29:1]   //29 - account cluster, change yours

delete from (select expand(accounts) from #30:0) where type = 'test2'  //30 - profile cluster, change yours
select from Account  // #29:0  and #29:1 not exist
select from Profile // accounts is [null,"#29:1"] in studio
```

### Runninng Mode
- [x] Remote

### OrientDB Version
- [x ] v2.1.x - Please specify last number: 2.1.17

### Operating System
- [ x] Linux ubuntu 14

### Java Version
- [ x] 8

**Comment by zella** [comment]
missprint, sorry: 2.1.17

**Comment by luigidellaquila** [comment]
Just pushed a fix on 2.1.x, the fix will be available in 2.1.18.

Now I'm porting it to 2.2

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Pushed on 2.2 as well, the fix will be released with 2.2.1 hotfix

Closing, thanks

Luigi
